packages/gatsby-source-filesystem/src/create-file-node.js

```diff
@@ -46,32 +46,37 @@ exports.createFileNode = async (
   }
 
   // Stringify date objects.
-  return JSON.parse(
-    JSON.stringify({
-      // Don't actually make the File id the absolute path as otherwise
-      // people will use the id for that and ids shouldn't be treated as
-      // useful information.
-      id: createNodeId(pathToFile),
-      children: [],
-      parent: null,
-      internal,
-      sourceInstanceName: pluginOptions.name || `__PROGRAMMATIC__`,
-      absolutePath: slashedFile.absolutePath,
-      relativePath: slash(
-        path.relative(
-          pluginOptions.path || process.cwd(),
-          slashedFile.absolutePath
-        )
-      ),
-      extension: slashedFile.ext.slice(1).toLowerCase(),
-      size: stats.size,
-      prettySize: prettyBytes(stats.size),
-      modifiedTime: stats.mtime,
-      accessTime: stats.atime,
-      changeTime: stats.ctime,
-      birthTime: stats.birthtime,
-      ...slashedFile,
-      ...stats,
-    })
-  )
+  const stats2 = { ...stats }
+  Object.keys(stats2).forEach(key => {
+    if (stats2[key] instanceof Date) {
+      stats2[key] = stats2[key].toJSON() // This is actually toISOString()
+    }
+  })
```

Comment:
shouldn't you use it? Also do we have to clone stats?

Reply:
Hum. Yes? :)

I'm cloning stats because I'm not comfortable modifying the result of .stat by reference. But I'm totally fine if you feel that's the right way to go. Shouldn't matter much for perf (a shallow clone is not the problem here).

Reply:
Doing it inline does appear to be (sliiightly yet consistently) faster

Reply:
Doing it explicitly is even better ("of course")
```
  stats.atime = stats.atime.toJSON()
  stats.mtime = stats.mtime.toJSON()
  stats.ctime = stats.ctime.toJSON()
  stats.birthtime = stats.birthtime.toJSON()
```

Reply:
I didn't see you using stats2, that was basically the point. I didn't care much about the cloning, it was more a question.